EIPS/eip-sc.md

```diff
@@ -0,0 +1,172 @@
+---
+eip: <to be assigned>
+title: Interface for Security Contact of Smart Contract
+description: An interface for security notice using asymmetric encryption
+author: Zainan Zhou (@xinbenlv)
+discussions-to: https://ethereum-magicians.org/t/erc-interface-for-security-contract/10303
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-08-09
+requires: 165
+---
+
+An interface for smart contract developer to specify to communicate to them in secured way.
+A typical use case is for security researcher to report to developer the
+security vulnerability.
+
+## Abstract
+An interface for security notice using asymmetric encryption. The interface exposes a asymmetric encryption key
+and a destination of delivery.
+
+## Motivation
+When security researchers identify security risk of a smart contract, currently there is no official way to
+report the problem to the smart contract maintainer, such as MakerDAO's issue.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+```solidity
+interface ERC_Security_Contact {
+
+    /// REQUIRED
+    function getSecurityContact(bytes memory data) public view
+    returns (
+        uint8 type,
+        bytes memory publicKey,
+        bytes memory extraData
+    );
+
+    /// OPTIONAL
+    function setSecurityContact(
+        uint8 type,
+        bytes memory publicKey,
+        bytes memory extraData) public;
+    event OnSetSecurityContact(uint8 type, bytes memory publicKeyForEncryption, bytes memory extraData);
+
+    /// OPTIONAL
+    function securityNotify(uint8 type, bytes memory data) public payable;
+    /// OPTIONAL
+    event OnSecurityNotification(uint8 type, bytes memory sourceData, uint256 value);
+
+    /// OPTIONAL
+    function bountyPolicy(uint id) public view returns(string);
+}
+```
+
+1. Complying interface MUST implement the method of `getSecurityContact`.
+
+The `type` is a one byte data with valid range of `[0x10, 0x7f]`. The ranges of `[0x00, 0x0f]` and `[0x80, 0xff]` are preserved for future extension.
+
+The `type` indicates the format of the `publicKey` and `extraData` in the following way
+
+------------------------------------------------------------------------------------------------
+| Type | Encryption scheme                   | extraData                                       |
+-------|-------------------------------------|--------------------------------------------------
+| 0x10 | GnuPG - RSA/3072                    | Email address(es) encoded in format of RFC 2822 |
```

Comment:
Would it also make sense to support a secp256k1 gpg algorithm?

Reply:
I was actually thinking of it! Because RSA keys were so long and inefficient. Added a TODO